libpod/runtime_img.go

```diff
@@ -123,8 +124,61 @@ func (r *Runtime) RemoveImage(image *image.Image, force bool) (string, error) {
 		// reponames and no force is applied, we error out.
 		return "", fmt.Errorf("unable to delete %s (must force) - image is referred to in multiple tags", image.ID())
 	}
+	err = image.Remove(force)
+	if err != nil && errors.Cause(err) == storage.ErrImageUsedByContainer {
+		errStorage := r.rmStorageContainers(force, image)
+		if errStorage == nil {
+			err = image.Remove(force)
```

Comment:
Are we retrying on the assumption that the containers using the image were removed? If so, can we get a comment to that effect?

Reply:
Also, instead of setting error, it's probably better to just `return` here and below in the else

Reply:
is there any chance that err might not be ErrImageUsedByContainer ?

Reply:
I've added a comment.

I think we want to capture the err from this second remove and return it if there's an error.  But if you really think it's better, I can just let the original err be returned.  As far as returning the errStorage error I was on the fence about returning that or not, but thought it would give a better clue if there was an error.  I'll keep them both for now, holler in he morn if you'd like me to remove them.

I think there's a very small chance that the err would be something other then ErrImageUsedByContainer.   But if it is, we'll just get the same error from the second image remove and will continue on down the error path.  But for the vast majority, it should be the image in use error.